app/integrations/discord/interactions/resolvers/ping.rb

```diff
@@ -0,0 +1,28 @@
+module Discord
+  module Interactions
+    module Resolvers
+      class Ping < Resolver
+        EXPECTED_GLOBAL_NAME = 'Discord'.freeze
+        PONG_RESPONSE_TYPE = 1
+
+        Response = Struct.new(:type)
+
+        def call
+          @response_status = RESPONSE_STATUS[:UNAUTHORIZED]
+          @body = {}
+
+          return unless valid_signature? && correct_global_name?
```

Comment:
do we need to validate this signature and the global name in all interactions or is it only needed on this `Ping` interaction?

Reply:
at `Resolver`, we'll have the `#call` method, which will be invoked for each child that needs it. `#call` will execute the validation, and if it's true, it will call the private `execute_action` method, which contains the logic for each Resolver child.


```ruby
module Discord
  module Interactions
    module Resolvers
      class Resolver
        ...
 
        def call
          if valid_signature?
            execute_action
          else
            unauthorized_response
          end
        end

         ...

        private

        attr_reader :request

        def execute_action
          raise NotImplementedError, "#{self.class} should implement 'call'"
        end

        def valid_signature?
          ...
        end
      end
    end
  end
end

```